File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack=new Stack<>();
        for(String t:tokens){
            if(!"+-*/".contains(t)){
                stack.push(Integer.parseInt(t));
            } else {
                int a=stack.pop();
                int b=stack.pop();
                switch (t){
                    case "+":stack.push(b+a); break;
                    case "-":stack.push(b-a); break;
                    case "*":stack.push(b*a); break;
                    case "/":stack.push((int)(b*1.0/a)); break;
                }
            }
        }
        return stack.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        stack = []
        for t in tokens:
            if t not in "+-*/":
                stack.append(int(t))
            else:
                a,b=stack.pop(),stack.pop()
                if t=="+":
                    stack.append(a+b)
                elif t=="-":
                    stack.append(b-a)
                elif t=="*":
                    stack.append(a*b)
                else:
                    stack.append(int(float(b)/a))
        return stack.pop()
        